src/main/kotlin/ui/BaseView.kt

```diff
@@ -1,24 +1,74 @@
 package ui
 
-import logic.exceptions.*
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import logic.exceptions.AuthenticationException
+import logic.exceptions.RetrievingDataFailureException
+import logic.exceptions.StoringDataFailureException
 import ui.cliPrintersAndReaders.CLIPrinter
 
-open class BaseView(
+abstract class BaseView(
```

Comment:
```suggestion
abstract class BaseView(
```
Maybe this name not good enough

I think using name that address what this class do like `performCLIReques` or `BaseRequestHandler`

Reply:
`RequestHandler` is good for me